joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForStatementsCreator.scala

```diff
@@ -100,6 +101,46 @@ trait AstForStatementsCreator(implicit withSchemaValidation: ValidationMode) { t
     controlStructureAst(ifNode, Some(notConditionAst), thenAst :: elseAsts)
   }
 
+  protected def astsForCaseExpression(node: CaseExpression): Seq[Ast] = {
+    def goCase(expr: Option[SimpleIdentifier]): List[RubyNode] = {
+      val elseThenClause: Option[RubyNode] = node.elseClause.map(_.asInstanceOf[ElseClause].thenClause)
+      val whenClauses                      = node.whenClauses.map(_.asInstanceOf[WhenClause])
+      val ifElseChain = whenClauses.foldRight[Option[RubyNode]](elseThenClause) {
```

Comment:
I only saw later that there is already a fold in the IfExpression handling, I will switch to that instead.

Reply:
Just a quick note about that fold - it was created when there was still no RubyNode and we were operating on top of ANTLR contexts. With dedicated RubyNodes for each construct, it might be more confusing than useful/elegant at times, so you are the better judge.

Reply:
Yeah, I later saw later I might have to do some weird special casing if I went that route for if vs elif, so this fold might be cleaner in this case. Just don't want to create a special snowflake in the the code.